### ISSUE TITLE
Unable to build for GTX1070 with CUDA 11 / cuDNN 8

### ISSUE THREAD

**Issue by MarvinKlemp** [AlexeyAB/darknet#6942]
OS: Ubuntu 18.04
CUDA: 11.0 from the install instructions
Compile with: make
Compile option: 
```
GPU=1
CUDNN=1
CUDNN_HALF=0
OPENCV=0
AVX=0
OPENMP=0
LIBSO=0
ZED_CAMERA=0
ZED_CAMERA_v2_8=0
```

```
nvcc --version

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2020 NVIDIA Corporation
Built on Mon_Oct_12_20:09:46_PDT_2020
Cuda compilation tools, release 11.1, V11.1.105
Build cuda_11.1.TC455_06.29190527_0
```

```
cat /usr/include/cudnn_version.h | grep CUDNN_MAJOR -A 2

#define CUDNN_MAJOR 8
#define CUDNN_MINOR 0
#define CUDNN_PATCHLEVEL 3
--
#define CUDNN_VERSION (CUDNN_MAJOR * 1000 + CUDNN_MINOR * 100 + CUDNN_PATCHLEVEL)

#endif /* CUDNN_VERSION_H */
```

```
nvidia-smi
Sun Nov  8 14:36:17 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 455.32.00    Driver Version: 455.32.00    CUDA Version: 11.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce GTX 1070    On   | 00000000:09:00.0  On |                  N/A |
|  0%   43C    P8    12W / 185W |    305MiB /  8118MiB |      1%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      1006      G   /usr/lib/xorg/Xorg                 18MiB |
|    0   N/A  N/A      1150      G   /usr/bin/gnome-shell               53MiB |
|    0   N/A  N/A      1463      G   /usr/lib/xorg/Xorg                112MiB |
|    0   N/A  N/A      1610      G   /usr/bin/gnome-shell               31MiB |
|    0   N/A  N/A      2078      G   ...AAAAAAAAA= --shared-files       83MiB |
+-----------------------------------------------------------------------------+
```

Im unable to compile it with a GTX 1070. If I understand correctly pascal architecture should still be supported by cuda 11. 
Any ideas what I can do ?

```
make

...

nvcc fatal   : Unsupported gpu architecture 'compute_30'
Makefile:177: recipe for target 'obj/convolutional_kernels.o' failed
make: *** [obj/convolutional_kernels.o] Error 1
```

**Comment by King-Key** [comment]
my gpu:GTX 2060,CUDA:11.0,it's also not suported,the question is cuda version?

**Comment by MarvinKlemp** [comment]
Just fixed it.

You should also explicitly set the arch flag.

```
// In my case of an gtx 1070:
ARCH= -gencode arch=compute_61,code=[sm_61,compute_61]
```

As the makefile right now has a lot of different archs. Also `30` which is not supported by CUDA 11 anymore.
Therefore I got the error

**Comment by MarvinKlemp** [comment]
closed view #6943 for more information
